### PR TITLE
Minor cleanups to image set compilation.

### PR DESCRIPTION
**Pull request by grp**
- Refactor some logic out into separate functions.
 - Cleanup code style nits for consistency.
 - Use filesystem to read JPEG images as well as PNG.
 - Move image set compilation into namespace and class.

### CODE REVIEW

**Comment by grp** [comment]
cc @marcsalem

File: Libraries/acdriver/Sources/CompileActionImageSet.cpp [hunk 1]
```diff
@@ -221,117 +210,222 @@ static bool readPNGFile(std::vector<unsigned char> &contents, std::string filena
...
+
+    uint32_t leftWidth = static_cast<uint32_t>(leftCapInset);
+    uint32_t centerWidth = static_cast<uint32_t>(width - (leftCapInset + rightCapInset));
+    uint32_t rightWidth = static_cast<uint32_t>(rightCapInset);
+
+    uint32_t topHeight = static_cast<uint32_t>(topCapInset);
+    uint32_t centerHeight = static_cast<uint32_t>(height - (topCapInset + bottomCapInset));
+    uint32_t bottomHeight = static_cast<uint32_t>(bottomCapInset);
+
+    uint32_t topVerticalOffset = static_cast<uint32_t>(height - bottomCapInset);
```
Comment (marcsalem) [inline Libraries/acdriver/Sources/CompileActionImageSet.cpp]: This is wrong, it should be:
uint32_t topVerticalOffset = static_cast<uint32_t>(height - topCapInset);

Comment (grp) [inline Libraries/acdriver/Sources/CompileActionImageSet.cpp]: Nice catch.

**Comment by facebook-github-bot-7** [comment]
Thank you for your pull request.  As you may know, we require contributors to sign our Contributor License Agreement, and we don't seem to have you on file and listed as active anymore.  In order for us to review and merge your code, please email cla@fb.com with your details so we can update your status.

**Comment by facebook-github-bot-6** [comment]
Thank you for your pull request.  As you may know, we require contributors to sign our Contributor License Agreement, and we don't seem to have you on file and listed as active anymore.  In order for us to review and merge your code, please email cla@fb.com with your details so we can update your status.

**Comment by grp** [comment]
@marcsalem, ping?

File: Libraries/acdriver/Sources/CompileActionImageSet.cpp [hunk 2]
```diff
@@ -29,183 +29,172 @@
...
         png_set_palette_to_rgb(png_struct_ptr);
         png_set_bgr(png_struct_ptr);
         png_set_filler(png_struct_ptr, 0xff, PNG_FILLER_AFTER);
-    }
-
-    if (color_type == PNG_COLOR_TYPE_GRAY &&
-        bit_depth < 8) {
+    } else if (color_type == PNG_COLOR_TYPE_GRAY && bit_depth < 8) {
         png_set_expand_gray_1_2_4_to_8(png_struct_ptr);
+    } else if (color_type == PNG_COLOR_TYPE_RGB || color_type == PNG_COLOR_TYPE_RGB_ALPHA) {
```
Comment (marcsalem) [inline Libraries/acdriver/Sources/CompileActionImageSet.cpp]: This is a regression. The code no longer transforms RGB files into RGBA.

We need the set_filler call to add the alpha channel, and this is now in the else statement when color_type is PNG_COLOR_TYPE_RGB.

Comment (grp) [inline Libraries/acdriver/Sources/CompileActionImageSet.cpp]: Good catch as well, fixed in update.

File: Libraries/acdriver/Sources/CompileActionImageSet.cpp [hunk 3]
```diff
@@ -29,183 +29,181 @@
...
+            png_set_bgr(png_struct_ptr);
+            png_set_filler(png_struct_ptr, 0xff, PNG_FILLER_AFTER);
+            break;
+        case PNG_COLOR_TYPE_RGB_ALPHA:
+            png_set_bgr(png_struct_ptr);
+            break;
     }
 
     if (bit_depth == 16) {
         png_set_strip_16(png_struct_ptr);
```
Comment (marcsalem) [inline Libraries/acdriver/Sources/CompileActionImageSet.cpp]: Apparently, the order of operations matters for some transforms, so I would keep the original order where the conversion to 8-bit is done first (before channel reordering, padding of alpha etc) unless you have tested the code with 16bit color PNGs.

**Comment by marcsalem** [comment]
I think if the transforms are returned to the original order, then this is good to go.

File: Libraries/acdriver/Sources/CompileActionImageSet.cpp (hunk 3 above)
Comment (grp) [inline Libraries/acdriver/Sources/CompileActionImageSet.cpp]: Done.

**Pull request merged**